dotnet/src/Connectors/Connectors.AI.Anthropic/AnthropicRequestSettings.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.AI;
+
+namespace Microsoft.SemanticKernel.Connectors.AI.Anthropic;
+
+/// <summary>
+/// Request settings for Claude.
+/// </summary>
+public class AnthropicRequestSettings : AIRequestSettings
+{
+    private int _maxTokensToSample = 256;
+    private readonly List<string> _stopSequences = new();
+    private double? _temperature = 1.0;
+    private double? _topP;
+    private int? _topK;
+
+    /// <summary>
+    /// The maximum number of tokens to generate before stopping.
+    /// </summary>
+    /// <remarks>
+    /// Note that our models may stop before reaching this maximum. This parameter only specifies the absolute maximum
+    /// number of tokens to generate.
+    /// </remarks>
+    [JsonPropertyName("max_tokens_to_sample")]
+    public int MaxTokensToSample
+    {
+        get => this._maxTokensToSample;
+        set
+        {
+            if (value < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(value), value, "MaxTokensToSample must be greater than zero.");
+            }
+
+            this._maxTokensToSample = value;
+        }
+    }
+
+    /// <summary>
+    /// Sequences that will cause the model to stop generating completion text.
+    /// </summary>
+    /// <remarks>
+    /// Our models stop on "\n\nHuman:", and may include additional built-in stop sequences in the future. By providing
+    /// the stop_sequences parameter, you may include additional strings that will cause the model to stop generating.
+    /// </remarks>
+    [JsonPropertyName("stop_sequences")]
+    public IList<string> StopSequences
+    {
+        get => this._stopSequences;
+        set
+        {
+            this._stopSequences.Clear();
+            this._stopSequences.AddRange(value);
```

Comment:
Setters like this shouldn't be copying; they should be storing the provided instance.